File: src/mod10_design_pattern/lesson9_facade/java/PreparingStudentFacade.java
package mod10_design_pattern.lesson9_facade.java;

public class PreparingStudentFacade {
    private BookShop bookShop;
    private BroomShop broomShop;
    private OwlShop owlShop;
    private WandShop wandShop;
    private TicketBooth ticketBooth;

    public PreparingStudentFacade(BookShop bookShop, BroomShop broomShop, OwlShop owlShop, WandShop wandShop, TicketBooth ticketBooth) {
        this.bookShop = bookShop;
        this.broomShop = broomShop;
        this.owlShop = owlShop;
        this.wandShop = wandShop;
        this.ticketBooth = ticketBooth;
    }


    public void prepareStudentToHogwarts() {
        bookShop.buyBooks();
        broomShop.buyBroom();
        owlShop.buyOwl();
        wandShop.buyWand();
        ticketBooth.buyTicket();
    }
}

File: src/mod10_design_pattern/lesson9_facade/kotlin/PreparingStudentFacade.kt
package mod10_design_pattern.lesson9_facade.kotlin

class PreparingStudentFacade {
    private val bookShop = BookShop()
    private val broomShop = BroomShop()
    private val wandShop = WandShop()
    private val owlShop = OwlShop()
    private val ticketBooth = TicketBooth()


    fun prepareStudentToHogwarts() {
        bookShop.buyBooks()
        broomShop.buyBroom()
        owlShop.buyOwl()
        wandShop.buyWand()
        ticketBooth.buyTicket()
    }
}